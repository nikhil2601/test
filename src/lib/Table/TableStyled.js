import styled from 'styled-components';

import { themeGet } from 'utils/theme';

const TableStyled = styled.table`
    border-collapse: collapse;
    overflow: hidden;
    overflow-x: auto;
    width: 100%;
    /**
     * Add all of the remaining styles from theme
     */
    ${themeGet('Table.styles')};
`;

export default TableStyled;
