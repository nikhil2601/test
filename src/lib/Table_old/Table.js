import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import { getThemeProps } from 'utils/theme';

export const TableContext = React.createContext({
    // The base url for making the internal API calls.
    baseUrl: '',
    // The different types of available table-cells.
    cellTypes: {},
    // The data to be rendered in the table.
    data: [],
    // The action handlers for the table.
    handlers: {},
    // The oAuth token for making the internal API calls.
    oAuthToken: '',
    // The sorting order.
    order: '',
    // The sorting key.
    orderBy: '',
    // The table schema.
    schema: {},
    // Add a selectable checkbox input for the row.
    selectable: null,
});

const TableStyled = styled.table`
    border-collapse: initial;
    border-radius: 5px;
    border-spacing: 0;
    border: 1px solid ${getThemeProps('palette.common.lighter')};
    display: table;
    overflow: hidden;
    overflow-y: visible;
    width: 100%;
    /**
     * Add all of the remaining styles from theme
     */
    ${getThemeProps('Table.styles')};
`;

const Table = ({ context, ...rest }) => (
    <TableContext.Provider value={{ ...context }}>
        <TableStyled {...rest} />
    </TableContext.Provider>
);

Table.propTypes = {
    /**
     * Add a context value to the Table and its children.
     */
    context: PropTypes.object,
};

Table.defaultProps = {
    context: {},
};

export default Table;
