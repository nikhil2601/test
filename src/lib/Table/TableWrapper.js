import styled, { css } from 'styled-components';

import Box from 'lib/Box';

import TableStyled from './TableStyled';

/**
 * A wrapper element around the Table component.
 *
 * @constructor
 */
const TableWrapper = styled(Box)`
    border-collapse: initial;
    margin: 0;
    overflow: visible;
    overflow-x: auto;
    padding: 0;
    /**
     * Add in dynamic styles based on column sizing.
     */
    ${TableStyled} {
        ${({ cols }) => {
            let minWidth = '';

            switch (cols) {
                case 1:
                case 2:
                    minWidth = '100%';
                    break;
                case 3:
                    minWidth = '600px';
                    break;
                case 4:
                    minWidth = '800px';
                    break;
                case 5:
                case 6:
                case 7:
                    minWidth = '900px';
                    break;
                case 8:
                    minWidth = '1000px';
                    break;
                case 9:
                    minWidth = '1100px';
                    break;
                case 10:
                    minWidth = '1200px';
                    break;
                case 11:
                    minWidth = '1300px';
                    break;
                case 12:
                    minWidth = '1400px';
                    break;
                default:
                    break;
            }

            return css`
                min-width: ${minWidth};
            `;
        }}
    }
`;

export default TableWrapper;
