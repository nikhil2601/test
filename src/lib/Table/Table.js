import PropTypes from 'prop-types';
import React from 'react';

import TableStyled from './TableStyled';
import TableWrapper from './TableWrapper';

/**
 * A simple `table` component.
 *
 * @constructor
 */
const Table = React.forwardRef((props, ref) => {
    const { boxProps, cols, ...rest } = props;

    return (
        <TableWrapper elevation={5} {...boxProps} cols={cols}>
            <TableStyled {...rest} cols={cols} ref={ref} />
        </TableWrapper>
    );
});

Table.displayName = 'Table';

Table.propTypes = {
    /**
     * The props applied to the inner `TableWrapper` component.
     */
    boxProps: PropTypes.object,
    /**
     * The amount of columns that are being rendered.
     */
    cols: PropTypes.number,
};

Table.defaultProps = {
    boxProps: null,
    cols: 12,
};

export default Table;
