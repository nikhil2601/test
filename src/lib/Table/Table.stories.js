import React from 'react';

import Table from './Table';
import TableHead from '../TableHead';
import TableRow from '../TableRow';
import TableBody from '../TableBody';
import TableCell from '../TableCell';

export default {
    title: 'Table',
    component: Table,
};

const tableData = {
    header: ['First name', 'Last name', 'Age', 'Foo', 'Foo'],
    body: ['John', 'Doe', '00', 'Foo', 'Foo'],
};

export const table = () => (
    <Table margin={2} placement="bottom-end" width="50%">
        <TableHead>
            <TableRow>
                {tableData.header.map(title => (
                    <TableCell>{title}</TableCell>
                ))}
            </TableRow>
        </TableHead>
        <TableBody>
            <TableRow>
                {tableData.body.map(value => (
                    <TableCell>{value}</TableCell>
                ))}
            </TableRow>
        </TableBody>
    </Table>
);

export const tableElevation = () => (
    <Table margin={2} cols={3} placement="bottom-end" width="50%">
        <TableHead>
            <TableRow>
                {tableData.header.map(title => (
                    <TableCell>{title}</TableCell>
                ))}
            </TableRow>
        </TableHead>
        <TableBody>
            <TableRow>
                {tableData.body.map(value => (
                    <TableCell>{value}</TableCell>
                ))}
            </TableRow>
        </TableBody>
    </Table>
);
