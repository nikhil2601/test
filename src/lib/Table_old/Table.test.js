import React from 'react';
import { shallow } from 'enzyme';

import Table from './index';

describe('<Table />', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<Table debug />);

        expect(component).toMatchSnapshot();
    });
});
