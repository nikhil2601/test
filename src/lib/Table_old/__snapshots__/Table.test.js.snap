// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render correctly in "debug" mode 1`] = `
<ContextProvider
  value={Object {}}
>
  <Table__TableStyled
    debug={true}
  />
</ContextProvider>
`;
